src/test/java/itest/ReportIT.java

```diff
@@ -0,0 +1,178 @@
+
+            webClient
+                    .post(RECORDING_REQ_URL)
+                    .sendForm(
+                            form,
+                            ar -> {
+                                if (assertRequestStatus(ar, postResponse)) {
+                                    MatcherAssert.assertThat(
+                                            ar.result().statusCode(), Matchers.equalTo(201));
+                                    postResponse.complete(null);
+                                }
+                            });
+
+            postResponse.get();
+
+            // Save the recording to archive
+            webClient
+                    .patch(String.format("%s/%s", RECORDING_REQ_URL, TEST_RECORDING_NAME))
+                    .sendBuffer(
+                            Buffer.buffer("SAVE"),
+                            ar -> {
+                                if (assertRequestStatus(ar, savedRecordingName)) {
+                                    MatcherAssert.assertThat(
+                                            ar.result().statusCode(), Matchers.equalTo(200));
+                                    MatcherAssert.assertThat(
+                                            ar.result()
+                                                    .getHeader(HttpHeaders.CONTENT_TYPE.toString()),
+                                            Matchers.equalTo(HttpMimeType.PLAINTEXT.mime()));
+                                    savedRecordingName.complete(ar.result().bodyAsString());
+                                }
+                            });
+
+            savedRecordingName.get();
+
+            // Get a report for the above recording
+            CompletableFuture<String> getResponse = new CompletableFuture<>();
+            webClient
+                    .get(String.format("%s/%s", REPORT_REQ_URL, savedRecordingName.get()))
+                    .send(
+                            ar -> {
+                                if (assertRequestStatus(ar, getResponse)) {
+                                    MatcherAssert.assertThat(
+                                            ar.result().statusCode(), Matchers.equalTo(200));
+                                    MatcherAssert.assertThat(
+                                            ar.result()
+                                                    .getHeader(HttpHeaders.CONTENT_TYPE.toString()),
+                                            Matchers.equalTo(HttpMimeType.HTML.mime()));
+                                    getResponse.complete(ar.result().bodyAsString());
+                                }
+                            });
+
+            int MAX_ERRORS_TRACKED = 100;
+            Parser parser = Parser.htmlParser().setTrackErrors(MAX_ERRORS_TRACKED);
+            Jsoup.parse(getResponse.get(), "", parser);
+            System.out.println(parser.getErrors());
+            MatcherAssert.assertThat(parser.getErrors().isEmpty(), Matchers.equalTo(true));
```

Comment:
`parser.getErrors()` should return an empty list if the report is a valid HTML document. I receive a list of errors instead. If I try to curl the same request, the HTML report renders correctly in my browser:
https://gist.github.com/jan-law/458bd4208344ba6cbe968d28d20df4ed

 I'm not sure if I'm passing the report response into the parser correctly, or even if `Jsoup` is an appropriate parser to use.
```
[INFO] Running itest.ReportIT
HTTP 404: Not Found
[9921: Unexpected token [EndTag] when in state [InBody], 10182: Unexpected token [EndTag] when in state [InBody], 13475: Unexpected token [EndTag] when in state [InBody], 15188: Unexpected token [EndTag] when in state [InBody], 17321: Unexpected token [EndTag] when in state [InBody], 21740: Unexpected token [EndTag] when in state [InBody], 26326: Unexpected token [EndTag] when in state [InBody], 32291: Unexpected token [EndTag] when in state [InBody], 35701: Unexpected token [EndTag] when in state [InBody], 41053: Unexpected token [EndTag] when in state [InBody], 42571: Unexpected token [EndTag] when in state [InBody], 45912: Unexpected token [EndTag] when in state [InBody], 48293: Unexpected token [EndTag] when in state [InBody], 57950: Unexpected token [EndTag] when in state [InBody], 62565: Unexpected token [EndTag] when in state [InBody], 72199: Unexpected token [EndTag] when in state [InBody], 74122: Unexpected token [EndTag] when in state [InBody], 87230: Unexpected token [EndTag] when in state [InBody], 90204: Unexpected token [EndTag] when in state [InBody], 108079: Unexpected token [EndTag] when in state [InBody], 115641: Unexpected token [EndTag] when in state [InBody]]
[ERROR] Tests run: 2, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 6.045 s <<< FAILURE! - in itest.ReportIT
[ERROR] testGetReportShouldSendFile  Time elapsed: 4.181 s  <<< FAILURE!
java.lang.AssertionError: 

Expected: <true>
     but: was <false>
	at itest.ReportIT.testGetReportShouldSendFile(ReportIT.java:134)
```

Reply:
It does look like you're using the correct JSoup HTML parser . What is the value of `getResponse.get()` at that point? Since it looks like there was a `404` first then the response body probably was not an HTML document, so the parser would choke on whatever it was instead - maybe just the text `Not Found`, for example. The parser errors output looks odd for that case since it seems to imply that it did parse a `<body>` or something - not sure.

Reply:
Printing `getResponse.get()` gives me an HTML file similar to the gist above. If I run the tests with `@TestMethodOrder`, the `404` appears after the parse error, which makes me think the `404` is from `testGetReportThrowsWithNonExistentRecordingName()`
```
[INFO] Running itest.ReportIT
<!--
   Copyright (c) 2018, Oracle and/or its affiliates. All rights reserved.
   Licensed under the Universal Permissive License v 1.0 as shown at http://oss.oracle.com/licenses/upl.
-->
<html lang="en">
<head>
<base target="_blank">
<meta http-equiv="X-UA-Compatible" content="IE=edge" />
<title>Automated Analysis Result Overview</title>
...
[9921: Unexpected token [EndTag] when in state [InBody], 
...
HTTP 404: Not Found
[ERROR] Tests run: 2, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 5.052 s
```

Reply:
Strange. What if you get rid of the `setTrackErrors` stuff and just use the plain `Jsoup.parse(file, "UTF-8")` like ClientAssetsIT does? Presumably, if the parser fails to parse the document, then the other test assertions would fail later on - things like simply checking that the parsed Document has a `<head>` and a `<body>`, again like the ClientAssetsIT does.

Reply:
Thanks, calling `Jsoup.parse(file, "UTF-8");` works well. I wrote the response into a temporary file in `test/resources` and deleted the file during cleanup.